File: android/ffmpeg-kit-android-lib/src/main/java/com/lgcns/ffmpegkit/StatisticsCallback.java
package com.lgcns.ffmpegkit;

/**
 * <p>Callback function that receives statistics generated for <code>FFmpegKit</code> sessions.
 */
@FunctionalInterface
public interface StatisticsCallback {

    /**
     * <p>Called when a statistics entry is received.
     *
     * @param statistics statistics entry
     */
    void apply(final Statistics statistics);

}

File: android/ffmpeg-kit-android-lib/src/main/java/com/lgcns/ffmpegkit/StatisticsCallback.kt
package com.lgcns.ffmpegkit

/**
 *
 * Callback function that receives statistics generated for `FFmpegKit` sessions.
 */
fun interface StatisticsCallback {
    /**
     *
     * Called when a statistics entry is received.
     *
     * @param statistics statistics entry
     */
    fun apply(statistics: Statistics)
}